### ISSUE TITLE
Need support for Docker Pro accounts

### ISSUE THREAD

**Issue by up_the_irons** [bacalhau-project/bacalhau#2230]
I raised the following issue in #bacalhau on Slack:

"I'm having issues getting bacalhau serve to "recognize" that I have a Docker Pro.  I do docker login as the same user running Docker, the auths key exists in Docker's config.json for the user and system-wide, yet I'm still seeing this error: `toomanyrequests: You have reached your pull rate limit.`"

Luke Marsden replied with:

"I wonder if this is because we use the docker API directly, please open a github issue :)"

So, here I am :)

**Comment by rossjones** [comment]
It looks like (from docker source) when directly using the API we need to pass an `AuthConfig` inside the `ImagePullOptions` to authenticate the image pull. This will mean we will need the docker hub creds for the user (or at least username and a token, in place of a password), and will need to be available to the compute node.

**Comment by simonwo** [comment]
Is this done @rossjones ?

**Comment by rossjones** [comment]
@simonwo Yes, all done. Specifying env vars for DOCKER_USERNAME/DOCKER_PASSWORD (use a read only token, not an actual password) will use those details when talking to docker hub for rate-limited things. Not sure where to document this yet though for compute providers.

**Comment by simonwo** [comment]
@fakela Can you suggest where to document this new feature for compute providers?
